src/vs/editor/contrib/gotoSymbol/browser/goToSymbol.ts

```diff
@@ -14,6 +15,20 @@ import { ITextModel } from 'vs/editor/common/model';
 import { ILanguageFeaturesService } from 'vs/editor/common/services/languageFeatures';
 import { ReferencesModel } from 'vs/editor/contrib/gotoSymbol/browser/referencesModel';
 
+function shouldIncludeLocationLink(sourceModel: ITextModel, loc: LocationLink): boolean {
+	// Always allow the location if the request comes from a document with the same scheme.
+	if (loc.uri.scheme === sourceModel.uri.scheme) {
+		return true;
+	}
+
+	// Otherwise filter out locations from internal schemes
+	if (matchesSomeScheme(loc.uri, Schemas.walkThroughSnippet, Schemas.vscodeChatCodeBlock)) {
+		return false;
```

Comment:
I have recently added `vscodeChatCodeCompreBlock` which I believe should also be blocked. We also have https://github.com/microsoft/vscode/blob/fd518e4db7bcf89d7e19bf29522786d0cd16a9a6/src/vs/platform/markers/common/markerService.ts#L15 which has some overlap but isn't the same. Maybe it is time to pull this into a single/shared place?